### DISCUSSION TITLE
Feedback on Insights App [1.2.0] 💭

### DISCUSSION THREAD

**Discussion by yahu1031** [serverpod/serverpod#1770] [Ideas]
#### Observation 1 (Enhancement Req) ####
As soon as the application was opened, it opened the manage projects dialog, which cannot be closed at any cost (neither Esc works nor outer tap). Showing the default screen would be a better idea instead of restricting the user or developer there. So the user or developer can explore the app much better before they start using it.

#### Observation 2 ####
When a project is selected and the server is not running, we get the `Details` button, which will show some error details stating we cannot connect to `address = localhost, port = <PORT>(some random 5-digit port)`. A minimal error message or piece of information will be better for the user experience. This goes with `Log Settings`, `Logs Screen`, `Health Screen`.

#### Observation 3 ####
The health metrics should be streamed every 5 or 1 sec. So that the developer or user doesn't need to keep refreshing the stats (maybe using a WebSocket). This will resolve one issue: the automatic disconnection of the server will be done without refreshing. This will optimize the app a little bit and make it super real-time.

#### Observation 4 ####
Providing full API call request and response data in the app will let the developer reduce the usage of devtools for network calls. Giving us log entries (which already exist) is an additional feature we have.

#### Observation 5 [🐛 Bug] ####
When I run the app, I can't get the list of endpoints and methods, even though I have created a new endpoint. The generator is watching, and the server is up.

I hope this feedback will add some value for further development in `serverpod_gui`(Serverpod App).

Thanks for the great product. Kudos to the whole team and contributors.

CC: @vlidholt, @SandPod, @Isakdl

**Comment by SandPod** [comment]
This is great, thank you for this super clear list  of feedback 🙏

We will take a look at it.
